app/controllers/api/distributions_controller.rb

```diff
@@ -14,4 +17,26 @@ def owner_resource
   def filtered(arel)
     polymorphic_filtered('distributable', arel)
   end
+
+  def create
+    create_resource.tap do |res|
+      consume! res, create_options
+      res = res.becomes(res.type.safe_constantize) if res.type
+      hal_authorize res
+      res.save!
+      distribute(res)
```

Comment:
I see what you're up to here... sort of a manual after-save-but-before-response hook.

The repeated HAL boilerplate above this does catch my attention.  Seems to be shared here and in the story-distributions controller.  Maybe relocate to a shared concern, along with the `decorator_for_model` stuff?